Add Config and Migration plus review changes

aphraoh opened the pull request:
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️

aphraoh committed:
Add Config and Migration, Update Adapter.php

aphraoh committed:
Update composer.json

aphraoh committed:
Update Adapter.php

aphraoh committed:
Update Adapter.php

aphraoh committed:
Update M240327150317Queue.php

aphraoh committed:
Update composer.json after review

aphraoh committed:
Use LoopInterface

aphraoh committed:
Use MutexInterface and MutexFactoryInterface

aphraoh committed:
Update src/Adapter.php

Co-authored-by: Dmitriy Derepko <xepozz@list.ru>

aphraoh committed:
Update composer.json

aphraoh committed:
Update composer.json

aphraoh committed:
Update Adapter.php

aphraoh committed:
Update composer.json

Build check error if not added to required

aphraoh committed:
Update composer.json due to invalid json

aphraoh commented:
Hi @vjik and @xepozz us requested except...

1. @xepozz Did not put "yiisoft/db-migration" to "suggested" section in composer.json as breaks build, please can we deal with this in another pull request.
2. @vjik not remove the empty di as this also cause problems (can not exactly remember what they were), please can we deal with this in another pull request.

samdark commented on config/di.php:
```diff
@@ -0,0 +1,6 @@
+
+declare(strict_types=1);
+
+return [
```
Does it make sense to provide the config if it is empty?

aphraoh replied:
Hi, have deleted it.

samdark commented on src/Adapter.php:
```diff
@@ -6,17 +6,24 @@
 {
     /**
-     * @var int timeout
+     * @var MutexInterface mutex interface
```
We're usually starting descriptions with capital letter and end these with dot. Not a huge issue though. Can fix it later.

```suggestion
     * @var MutexInterface Mutex interface.
```

samdark commented on src/Adapter.php:
```diff
@@ -92,7 +97,7 @@ public function push(MessageInterface $message): MessageInterface
     public function subscribe(callable $handlerCallback): void
     {
-        $this->runExisting($handlerCallback);
+        $this->run($handlerCallback, true, 5); // TWK TODO timeout should not be hard coded
```
Do you plan to resolve it before merge?

aphraoh replied:
Can I do it after the merge, I believe the solution should be somewhere in the middleware but need to investigate a bit further so maybe that can be a lession for another day after the merge...

samdark commented on src/Adapter.php:
```diff
@@ -115,10 +121,10 @@ public function withChannel(string $channel): self
-        // TWK TODO ??? if (!$this->mutex->acquire(__CLASS__ . $this->channel, $this->mutexTimeout)) {
-        // TWK TODO ??? throw new \Exception('Has not waited the lock.');
-        // TWK TODO ??? }
+        // TWK TODO what is useMaster in Yii3 return $this->db->useMaster(function () {
```
@Tigrov, @terabytesoftw?

samdark commented on src/Adapter.php:
```diff
@@ -194,4 +200,28 @@ private function moveExpired(): void
+    { 
+        while ($this->loop->canContinue()) {
+            if ($payload = $this->reserve()) {
+                if ($handlerCallback(\unserialize($payload['job']))) {
```
In general, we're not using PHP serialization for queue. Is it necessary?

xepozz commented on src/Adapter.php:
```diff
@@ -194,4 +200,28 @@ private function moveExpired(): void
+    { 
+        while ($this->loop->canContinue()) {
+            if ($payload = $this->reserve()) {
+                if ($handlerCallback(\unserialize($payload['job']))) {
```
Use https://github.com/yiisoft/queue/blob/master/src/Message/MessageSerializerInterface.php instead

aphraoh replied:
OK, will use this interface

xepozz commented:
@aphraoh Do not remove discussions, do not use force-push, just add commits one by one, do not care about the rest.

As I wrote before, I don't like `yiisoft/db-migration` and `yiisoft/mutex` in `require` section. If the second is used in the adapter, then first is just declared one files with sql and it maybe useless. 

Move it to `suggests` composer section.

samdark commented:
> just add commits one by one, do not care about the rest.

Yes. We take care about squashing commits when merging, but we need to see how it progressed while doing review. Also, force-pushing is quite confusing for GitHub UI.

aphraoh committed:
Delete config directory due to empty di

aphraoh committed:
Update src/Adapter.php

Start description with capital letter and end with dot.

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>

aphraoh commented:
> > just add commits one by one, do not care about the rest.
> 
> Yes. We take care about squashing commits when merging, but we need to see how it progressed while doing review. Also, force-pushing is quite confusing for GitHub UI.

Oh dear, please send me a link to the process I should following if I would like to ammend a pull request in github :-|

aphraoh committed:
Delete src/Migrations directory

aphraoh committed:
Update composer.json

aphraoh committed:
Update Adapter.php

Update var comments to convention
Use serialize interface instead of php serialise

aphraoh commented:
Hi @xepozz @samdark @vjik , hopefully we can merge with the following outstanding...

1. // TWK TODO timeout should not be hard coded
2. // TWK TODO what is useMaster in Yii3
3. @xepozz, have removed yiisoft/db-migration from composer.json, but yiisoft/mutex still in required section cause used in Adopter (see constructor)

aphraoh commented:
Apologies in advance if I have done any squeezed and forced commits, still googling to see exactly what you mean by that...

samdark commented:
@aphraoh latest commits are fine.

samdark commented:
Let's merge it and address comments separately. Good job, @aphraoh.

aphraoh commented:
Sent from my iPhoneOn 3 Apr 2024, at 17:33, Alexander Makarov ***@***.***> wrote:﻿
Let's merge it and address comments separately. Good job, @aphraoh.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
